feed/obs/thread.js

```diff
@@ -23,11 +27,25 @@ exports.create = function (api) {
 
     var prepend = MutantArray()
     api.sbot.async.get(rootId, (err, value) => {
-      sync.set(true)
       if (!err) {
         var msg = unboxIfNeeded({key: rootId, value})
         if (isBlocked(msg)) msg.isBlocked = true
-        prepend.push(Value(msg))
+
+        if (isBlog(msg)) {
+          // resolve the blog body before returning
+          Blog(api.sbot.obs.connection).async.get(msg, (err, result) => {
+            if (!err) {
+              msg.body = result.body
```

Comment:
heads up - we're following the _decorator pattern_ - the result is just a sugared message, so you could over-write msg with result and the key and value and body would all be there. Not that relevant here but so you know

Reply:
Hmmm, I think this is a reminder that feed.obs.thread needs some serious work.

Reply:
This way preserves compat with existing code.